src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java

```diff
@@ -17,11 +20,24 @@
     private String givenName;
 
     public void validate() {
+        Pattern p = Pattern.compile("[!@#\\r\\n$%¶*\\\\()_:/+=|<>?{}\\[\\]~-]");
+
+
         if (!StringUtils.hasText(givenName) || givenName.length() > MAX_STRING_LENGTH) {
             throw new CreateCertificateException(INVALID_GIVEN_NAME);
         }
         if (!StringUtils.hasText(familyName) || familyName.length() > MAX_STRING_LENGTH) {
             throw new CreateCertificateException(INVALID_FAMILY_NAME);
         }
+
+        Matcher givenNameMatcher = p.matcher(givenName);
+        Matcher familyNameMatcher = p.matcher(familyName);
```

Comment:
Put the creation of the familyMatcher after the if of givenNameMatcher (before line 39). If the givenNameMatcher.find() succeeds the familyNameMatcher is not needed anymore.